src/celeritas/global/detail/AlongStepActionImpl.hh

```diff
@@ -145,15 +148,20 @@ inline CELER_FUNCTION void along_step_track(CoreTrackView const& track)
         const auto& urban_data = phys.urban_data();
 
         // Replace step with actual geometry distance traveled
-        auto msc_step_result      = track.make_physics_step_view().msc_step();
-        msc_step_result.geom_path = geo_step;
+        auto msc_step_result = track.make_physics_step_view().msc_step();
+        if (geo_limited)
+        {
+            msc_step_result.geom_path    = geo_step;
+            msc_step_result.is_displaced = false;
```

Comment:
We should probably revisit how the "is_displaced" and "geo_limited" and "action result" complement each other: they're not orthogonal choices now.

Reply:
Can I leave this to you once you finish refactoring the along step action?

Reply:
Yep! Just sort of a note to self.